src/phoenix/experimental/evals/models/openai.py

```diff
@@ -175,23 +307,28 @@ def _generate_with_retry(self, **kwargs: Any) -> Any:
             self._openai_error.ServiceUnavailableError,
         ]
 
+        async def metered_openai_completion(**kwargs: Any) -> Any:
+            limit = self.rate_limiter.alimit(self.model_name, openai_token_usage)
+            response = await limit(self._openai.ChatCompletion.acreate)(**kwargs)
+            return response
+
         @self.retry(
             error_types=openai_retry_errors,
             min_seconds=self.retry_min_seconds,
             max_seconds=self.retry_max_seconds,
             max_retries=self.max_retries,
         )
-        def _completion_with_retry(**kwargs: Any) -> Any:
+        async def _completion_with_retry(**kwargs: Any) -> Any:
             if self._model_uses_legacy_completion_api:
                 if "prompt" not in kwargs:
                     kwargs["prompt"] = "\n\n".join(
                         (message.get("content") or "")
                         for message in (kwargs.pop("messages", None) or ())
                     )
-                return self._openai.Completion.create(**kwargs)
-            return self._openai.ChatCompletion.create(**kwargs)
+                return await metered_openai_completion(**kwargs)
+            return await metered_openai_completion(**kwargs)
```

Comment:
Are these two lines supposed to be the same? Are we still handling the legacy completion API somewhere?
```suggestion
            return await metered_openai_completion(**kwargs)
```

Reply:
This is a drop-in replacement for the existing codepaths. I am avoiding  refactors of existing code while we add new concepts

Reply:
Then the drop-in replacement would have to be like the following, so that the legacy completion api is handled separately.
```suggestion
                return await metered_openai_completion(**kwargs)
            return await metered_openai_chat_completion(**kwargs)
```

Reply:
gotcha, I understand